### PR TITLE
FHB-549 : Database Schema into Shared Kernel

### PR DESCRIPTION
**Pull request by Zac-AND-Digital**
Ticket: [FHB-549](https://dfedigital.atlassian.net/browse/FHB-549)

---

Developed & Tested on: https://github.com/DFE-Digital/fh-services/pull/92

---

This PR creates the relationships, navigations and updates the entities so that we can take a deserialized Service and write it into the database, as well as read it out of the Db into a Service object later.

I implemented all of this on https://github.com/DFE-Digital/fh-services/pull/92 and tested the full journey as working:

- JSON from the Mock API deserializes into a fully nested `Service`
- The Service object can then be added to the database and `SaveChangesAsync()` can be called
- The database successfully contains the fully mapped `Service` with all its relationships
- When reading from the Db, e.g., with `Service? serviceFromDb = await dbContext.Services.Where(x => x.OrId == "<UUID>").FirstOrDefaultAsync()`, the `Service` is returned with all nested relationships intact.
- The returned `Service` can then serialized into JSON, and that JSON contains intact all the information that originally came from the mock API that was inserted in the first place.

After this ticket is merged, the Service Directory API will then use this package and add a new migration, and I will edit https://github.com/DFE-Digital/fh-services/pull/92 to use this package :)

[FHB-549]: https://dfedigital.atlassian.net/browse/FHB-549?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs [hunk 1]
```diff
@@ -1,87 +1,59 @@
...
-    public string? Freq { get; init; }
+    [JsonPropertyName("freq")] public string? Freq { get; init; }
 
-    [JsonPropertyName("interval")]
-    public short? Interval { get; init; }
+    [JsonPropertyName("interval")] public short? Interval { get; init; }
 
-    [JsonPropertyName("byday")]
-    public string? Byday { get; init; }
+    [JsonPropertyName("byday")] public string? Byday { get; init; }
```
Comment (stevesatdfe) [inline src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs:34]: Should that be ByDay for pascal format?

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs [hunk 2]
```diff
@@ -1,87 +1,59 @@
...
 
-    [JsonPropertyName("byweekno")]
-    public string? Byweekno { get; init; }
+    [JsonPropertyName("byweekno")] public string? Byweekno { get; init; }
```
Comment (stevesatdfe) [inline src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs:36]: Should that be ByWeekNo for pascal format? Same for others below

**Review by Top-Cat — APPROVED** [review]

**Review by stevesatdfe — CHANGES_REQUESTED** [review]

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs (hunk 1 above)
Comment (Top-Cat) [reply src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs]: Assume they're to match the casing in the spec
https://github.com/openreferral/specification/blob/3.0/schema/schedule.json#L172

**Review by Top-Cat — APPROVED** [review]

File: src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs (hunk 2 above)
Comment (Zac-AND-Digital) [reply src/shared/shared-kernel/src/fh-shared-kernel.shared-kernel/OpenReferral/Entities/Schedule.cs]: Resolved in: [732d3da](https://github.com/DFE-Digital/fh-services/pull/111/commits/732d3da30f45cb57f65a875bc2f6713b690e07ee)

**Review by stevesatdfe — APPROVED** [review]

**Pull request merged**